src/pages/lessons/[slug].tsx

```diff
@@ -432,41 +441,79 @@ const Lesson: React.FC<LessonProps> = ({initialLesson}) => {
               send(`RECOMMEND`)
             }
           })
-
         break
     }
-  }, [currentPlayerState, session_id])
+  }, [currentLessonState, session_id])
 
   React.useEffect(() => {
-    // when this effect is triggered, put the machine in the loading state (via
-    // `LOAD`). That state should take care of the rest with an Invoked
-    // Callback.
-    send({
-      type: 'LOAD',
-      lesson: initialLesson,
-      viewer,
+    // Keep lesson machine state in sync with
+    // videoService to control overlays and stuff
+    if (!isWaiting) {
+      hasEnded && send('COMPLETE')
+      isPaused ? send('PAUSE') : send('PLAY')
+    }
+  }, [hasEnded, isPaused, isWaiting])
```

Comment:
This is the bridge between player state and lesson state. We mostly just want to control overlays which happen to be tied to video `hasEnded` or `isWaiting` states. So this should be all we need at the moment.